geth/transactions/queue/queue.go

```diff
@@ -176,118 +148,78 @@ func (q *TxQueue) Enqueue(tx *common.QueuedTx) error {
 
 	// notify handler
 	log.Info("calling txEnqueueHandler")
-	q.txEnqueueHandler(tx)
-
 	return nil
 }
 
 // Get returns transaction by transaction identifier
 func (q *TxQueue) Get(id common.QueuedTxID) (*common.QueuedTx, error) {
-	q.mu.RLock()
-	defer q.mu.RUnlock()
+	q.mu.Lock()
+	defer q.mu.Unlock()
 
 	if tx, ok := q.transactions[id]; ok {
+		if _, inprogress := q.inprogress[id]; inprogress {
+			return tx, ErrQueuedTxInProgress
+		}
+		q.inprogress[id] = empty{}
```

Comment:
Why do we set the transaction to `inprogress` map here in `Get` method? Getters should not perform any side effects.

Reply:
I noticed that in the current code there is only one place where Get is called (CompleteTransaction), so I wanted to get rid of the methods that are not used

Reply:
or better to say redundant

Reply:
yeah, it doesn't make sense i will add separate method for Get and another for marking transaction as inprogress